scripts/exregional_run_analysis.sh

```diff
@@ -0,0 +1,594 @@
+In directory:     \"${scrfunc_dir}\"
+
+This is the ex-script for the task that runs a analysis with FV3 for the
+specified cycle.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Specify the set of valid argument names for this script/function.  
+# Then process the arguments provided to this script/function (which 
+# should consist of a set of name-value pairs of the form arg1="value1",
+# etc).
+#
+#-----------------------------------------------------------------------
+#
+valid_args=( "CYCLE_DIR" "ANALWORKDIR" "CYCLE_ROOT")
+process_args valid_args "$@"
+#
+#-----------------------------------------------------------------------
+#
+# For debugging purposes, print out values of arguments passed to this
+# script.  Note that these will be printed out only if VERBOSE is set to
+# TRUE.
+#
+#-----------------------------------------------------------------------
+#
+print_input_args valid_args
+#
+#-----------------------------------------------------------------------
+#
+# Load modules.
+#
+#-----------------------------------------------------------------------
+#
+case $MACHINE in
+#
+"WCOSS_C" | "WCOSS")
+#
+
+  if [ "${USE_CCPP}" = "TRUE" ]; then
+  
+# Needed to change to the experiment directory because the module files
+# for the CCPP-enabled version of FV3 have been copied to there.
+
+    cd_vrfy ${CYCLE_DIR}
+  
+    set +x
+    source ./module-setup.sh
+    module use $( pwd -P )
+    module load modules.fv3
+    module list
+    set -x
+  
+  else
+  
+    . /apps/lmod/lmod/init/sh
+    module purge
+    module use /scratch4/NCEPDEV/nems/noscrub/emc.nemspara/soft/modulefiles
+    module load intel/16.1.150 impi/5.1.1.109 netcdf/4.3.0 
+    module list
+  
+  fi
+
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="mpirun -l -np ${PE_MEMBER01}"
+  ;;
+#
+"THEIA")
+#
+
+  if [ "${USE_CCPP}" = "TRUE" ]; then
+  
+# Need to change to the experiment directory to correctly load necessary 
+# modules for CCPP-version of FV3SAR in lines below
+    cd_vrfy ${EXPTDIR}
+  
+    set +x
+    source ./module-setup.sh
+    module use $( pwd -P )
+    module load modules.fv3
+    module load contrib wrap-mpi
+    module list
+    set -x
+  
+  else
+  
+    . /apps/lmod/lmod/init/sh
+    module purge
+    module use /scratch4/NCEPDEV/nems/noscrub/emc.nemspara/soft/modulefiles
+    module load intel/16.1.150 impi/5.1.1.109 netcdf/4.3.0 
+    module load contrib wrap-mpi 
+    module list
+  
+  fi
+
+  ulimit -s unlimited
+  ulimit -a
+  np=${SLURM_NTASKS}
+  APRUN="mpirun -np ${np}"
+  ;;
+#
+"HERA")
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  LD_LIBRARY_PATH="${UFS_WTHR_MDL_DIR}/FV3/ccpp/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+  ;;
+#
+"JET")
+  ulimit -s unlimited
+  ulimit -a
+  APRUN="srun"
+  LD_LIBRARY_PATH="${UFS_WTHR_MDL_DIR}/FV3/ccpp/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+  NCKS=/apps/nco/4.1.0-intel/bin/ncks
```

Comment:
Does NCKS not exist in a module file? Can this be moved to the module file that we are managing here so we keep all the requirements in the same file?

Reply:
Only certain versions of the NCKS works on JET so far. I prefer to leave it as is now as I don't know if I can make right change now. Still need to figure out how to control the environmental for GSI runs.